### PR TITLE
ISPN-6429 Duplicated boolean projections are broken in compat mode

### PR DESCRIPTION
**Pull request by anistor**
https://issues.jboss.org/browse/ISPN-6429

### CODE REVIEW

File: server/integration/feature-pack/src/main/resources/modules/system/layers/base/org/infinispan/objectfilter/main/module.xml [hunk 1]
```diff
@@ -9,7 +9,6 @@
       <module name="org.infinispan.query.dsl"/>
       <module name="org.antlr.antlr-runtime"/>
       <module name="org.hibernate.hql.parser"/>
-      <module name="org.antlr.antlr-runtime"/>
```
Comment (slaskawi) [inline server/integration/feature-pack/src/main/resources/modules/system/layers/base/org/infinispan/objectfilter/main/module.xml]: Is this intentional? If so, perhaps we could delete `antlr` from some WF modules (`as-modules/*`)?

**Comment by slaskawi** [comment]
The changes look good to me. You could squash all the commits though (but it's a minor thing)

File: server/integration/feature-pack/src/main/resources/modules/system/layers/base/org/infinispan/objectfilter/main/module.xml (hunk 1 above)
Comment (anistor) [inline server/integration/feature-pack/src/main/resources/modules/system/layers/base/org/infinispan/objectfilter/main/module.xml]: We cannot delete antlr, unfortunatelly. But here it was imported twice so I removed one.

**Comment by anistor** [comment]
I'll squash the two commits related to ISPN-6429. I initially chose to separate the test and the fix so the reviewer can easily revert the fix and see the test failing.

File: server/integration/feature-pack/src/main/resources/modules/system/layers/base/org/infinispan/objectfilter/main/module.xml (hunk 1 above)
Comment (slaskawi) [inline server/integration/feature-pack/src/main/resources/modules/system/layers/base/org/infinispan/objectfilter/main/module.xml]: ooh yeah - I'm blind... sorry

**Comment by slaskawi** [comment]
Ok, great! Thanks!

**Comment by anistor** [comment]
Updated.

**Comment by slaskawi** [comment]
pushed thanks!

**Pull request closed without merging**